sdks/opik_optimizer/src/opik_optimizer/meta_prompt_optimizer.py

```diff
@@ -0,0 +1,284 @@
+            print(f"Current best score: {best_score:.4f}")
+
+            # Generate new prompts using reasoning
+            new_prompts = self._generate_prompts(current_prompt, current_score)
+            
+            # Evaluate new prompts
+            prompt_scores = []
+            for new_prompt in new_prompts:
+                score = self._evaluate_prompt(new_prompt, **kwargs)
+                improvement = (score - current_score) / current_score if current_score > 0 else 0
+                prompt_scores.append({
+                    "prompt": new_prompt,
+                    "score": score,
+                    "improvement": improvement
+                })
+                print(f"Generated prompt score: {score:.4f}")
+
+            # Find best prompt from this round
+            best_round_prompt = max(prompt_scores, key=lambda x: x["score"])
+            
+            # Check if we've improved enough
+            improvement = (best_round_prompt["score"] - best_score) / best_score if best_score > 0 else 0
+            if improvement < self.improvement_threshold:
+                print(f"Improvement ({improvement:.2%}) below threshold ({self.improvement_threshold:.2%}). Stopping.")
+                stopped_early = True
+                break
+
+            # Update best prompt if improved
+            if best_round_prompt["score"] > best_score:
+                best_prompt = best_round_prompt["prompt"]
+                best_score = best_round_prompt["score"]
+                current_prompt = best_round_prompt["prompt"]
+                current_score = best_round_prompt["score"]
+                print(f"New best prompt found with score: {best_score:.4f}")
+
+            # Record round results
+            rounds.append(OptimizationRound(
+                round_number=round_num + 1,
+                current_prompt=current_prompt,
+                current_score=current_score,
+                generated_prompts=[{
+                    "prompt": p["prompt"],
+                    "score": p["score"],
+                    "improvement": p["improvement"]
+                } for p in prompt_scores],
+                best_prompt=best_round_prompt["prompt"],
+                best_score=best_round_prompt["score"],
+                improvement=improvement
+            ))
+
+        # If no rounds were completed, add the initial round
+        if not rounds:
+            rounds.append(OptimizationRound(
+                round_number=1,
+                current_prompt=current_prompt,
+                current_score=current_score,
+                generated_prompts=[{
+                    "prompt": current_prompt,
+                    "score": current_score,
+                    "improvement": 0.0
+                }],
+                best_prompt=best_prompt,
+                best_score=best_score,
+                improvement=0.0
+            ))
+
+        return OptimizationResult(
```

Comment:
Should have the same results as the other optimizers.

We'll add a get_history() method to BaseOptimizer to get the other details (initial values, etc)